domain/status/state/state.go

```diff
@@ -242,6 +242,105 @@ ON CONFLICT(application_uuid) DO UPDATE SET
 	return nil
 }
 
+// GetRelationStatus gets the status of the given relation. It returns an error
+// satisfying [statuserrors.RelationNotFound] if the relation doesn't exist.
+func (st *State) GetRelationStatus(
```

Comment:
todo: make a note in the jira card to ensure everything is enabled at the end of the epic to double check if this method is required and remove if not.

Reply:
I've actually ended up making this private to the package anyway, since I've moved the the get into the set state layer set, rather than the service layer. Guilluame pointed out that we don't have proper transactionality the way it was.